lib/internal/Magento/Framework/App/HttpRequestInterface.php

```diff
@@ -0,0 +1,52 @@
+<?php
```

Comment:
Maybe you have this [class](https://github.com/magento/zf1/blob/master/library/Zend/Controller/Request/Http.php) for this purpose.

Reply:
This interface extends the basic interface and contains methods for determining which HTTP request used. I did not touch the original interface to not do BC break. But if we can move this "contracts" to original Request interface(https://github.com/magento/magento2/blob/56af1e73ce21867b770a7458ab6e109f4a1eface/lib/internal/Magento/Framework/App/RequestInterface.php) it is even better. Because then it will not be necessary to make checks for implementation or special annotations like "$request instance of HttpRequestInterface"  or "/** @var $request HttpRequestInterface $request */"

Reply:
Thanks! In this case we must create new Interface to not do BC break as you say!